commands/cloudapi-v6/image.go

```diff
@@ -104,9 +109,274 @@ func ImageCmd() *core.Command {
+		if resp != nil && printer.GetId(resp) != "" {
+			c.Printer.Verbose(config.RequestInfoMessage, printer.GetId(resp), resp.RequestTime)
+		}
+		if err != nil {
+			return err
+		}
+		if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
+			return err
+		}
+		return c.Printer.Print(getImagePrint(resp, c, nil))
+	}
+}
+
+// Util func - Given a slice of public & non-public images, return only those images that are non-public.
+// If any image in the slice has null properties, or "Properties.Public" field is nil, an error is thrown.
+func _getNonPublicImages(imgs *[]ionoscloud.Image) (*[]ionoscloud.Image, error) {
+	var nonPublicImgs []ionoscloud.Image
+	for _, i := range *imgs {
+		properties, ok := i.GetPropertiesOk()
+		if !ok {
+			return nil, fmt.Errorf("properties of image %s are nil\n", *i.GetId())
+		}
+		isPublic, ok := properties.GetPublicOk()
+		if !ok {
+			return nil, fmt.Errorf("public field of image %s is nil\n", *i.GetId())
+		}
+		if !*isPublic {
+			nonPublicImgs = append(nonPublicImgs, i)
+		}
+	}
+	return &nonPublicImgs, nil
+}
+
+//  deletes non-public images, as deleting public images is forbidden by the API.
+func DeleteAllNonPublicImages(c *core.CommandConfig) error {
+	depth := int32(1)
+	images, resp, err := c.CloudApiV6Services.Images().List(
+		resources.ListQueryParams{QueryParams: resources.QueryParams{Depth: &depth}},
+	)
+	if err != nil {
+		return err
+	}
+	items, ok := images.GetItemsOk()
+	if !(ok && len(*items) > 0 && items != nil) {
+		return errors.New("could not retrieve images")
+	}
+
+	if items, err = _getNonPublicImages(items); err != nil {
+		return err
+	}
+	if len(*items) < 1 {
+		return errors.New("no non-public images found")
+	}
+
+	_ = c.Printer.Warn("Images to be deleted:")
+	// TODO: this is duplicated across all resources - refactor this (across all resources)
+	for _, img := range *items {
+		delIdAndName := ""
+		if id, ok := img.GetIdOk(); ok && id != nil {
+			delIdAndName += "ID: `" + *id
+		}
+		if properties, ok := img.GetPropertiesOk(); ok && properties != nil {
+			if name, ok := properties.GetNameOk(); ok && name != nil {
+				delIdAndName += "`, Name: " + *name
+			}
+		}
+		_ = c.Printer.Warn(delIdAndName)
+	}
+
+	if err = utils.AskForConfirm(c.Stdin, c.Printer, "delete all the images"); err != nil {
+		return err
+	}
+	c.Printer.Verbose("Deleting all the images...")
+
+	var multiErr error
+	for _, img := range *items {
+		if id, ok := img.GetIdOk(); ok && id != nil {
+			c.Printer.Verbose("Starting deleting image with id: %v...", *id)
+			resp, err = c.CloudApiV6Services.Images().Delete(*id, resources.QueryParams{})
+			if resp != nil && printer.GetId(resp) != "" {
+				c.Printer.Verbose(config.RequestInfoMessage, printer.GetId(resp), resp.RequestTime)
+			}
+			if err != nil {
+				multiErr = multierr.Append(multiErr, fmt.Errorf(config.DeleteAllAppendErr, c.Resource, *id, err))
+				continue
+			} else {
```

Comment:
doesn't seem you need an else for the print

Reply:
i know, right? This is the case for all resources. At first I wanted to change this, but kept it to be the same behaivour across all resources.

One more thing, I kept it the same as for other resources because when we eventually refactor this code, we can then do a find & replace with regex match.